Steer/src/CollisionContextTool.cxx

```diff
@@ -245,12 +247,15 @@ int main(int argc, char* argv[])
         sampler.setBunchFilling(options.bcpatternfile);
       }
       auto orbitstart = options.tfid * options.orbitsPerTF;
-      sampler.setFirstIR(o2::InteractionRecord(0, orbitstart));
-      sampler.init();
       o2::InteractionTimeRecord record;
-      int count = 0;
+      // this loop makes sure that we have at least one collision within the number of orbits asked (if noEmptyTF is enabled)
       do {
+        sampler.setFirstIR(o2::InteractionRecord(0, orbitstart));
+        sampler.init();
         record = sampler.generateCollisionTime();
+      } while (options.noEmptyTF && usetimeframelength && record.orbit >= orbitstart + options.orbits);
+      int count = 1;
+      do {
```

Comment:
no ... since this only applies when I am in "usetimeframelength" mode. In this case, the number of events should be determined dynamically and mcnumberasked neglected.